### PR TITLE
Added sorting items, refactored code

### PR DESCRIPTION
**Pull request by volodymyryaniv**

### CODE REVIEW

File: client/src/redux/selectors/todolistSelectors/index.ts [hunk 1]
```diff
@@ -22,3 +30,85 @@ export const selectCompleted = createSelector(
...
+    case 'expiry-date-asc':
+      return list.sort(expiryAsc);
+    case 'expiry-date-desc':
+      return list.sort(expiryDesc);
+    case 'text-asc':
+      return list.sort(textAsc);
+    case 'text-desc':
+      return list.sort(textDesc);
+    default:
+      return list;
```
Comment (Yuliia143) [inline client/src/redux/selectors/todolistSelectors/index.ts]: Try to work with enum for orders- create it in separate file and use them

File: client/src/modules/SortList/SortList.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  useEffect(() => {
+    const closeSortList = (e: MouseEvent) => {
+      const path = e.composedPath();
+      if (ref.current) {
+        if (!path.includes(ref.current)) {
+          setShowSort(false);
+        }
+      }
+    };
```
Comment (Yuliia143) [inline client/src/modules/SortList/SortList.tsx]: You can take function out of useEffect, you can describe it in upper scope. Keep useEffect clean also

File: client/src/modules/SortList/SortList.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          className={icon}
+          src="/icons/sort.png"
+          alt="sort icon"
+          onClick={() => setShowSort((prev) => !prev)}
+        />
+      </div>
+      {showSort && (
+        <ul className={container}>
+          {categories.map((category) => {
+            return (
```
Comment (Yuliia143) [inline client/src/modules/SortList/SortList.tsx]: You can skip 'return'

File: client/src/redux/selectors/todolistSelectors/index.ts [hunk 4]
```diff
@@ -22,3 +30,85 @@ export const selectCompleted = createSelector(
...
+  const first = new Date(a.expireUntil);
+  const second = new Date(b.expireUntil);
+  return first > second ? 1 : -1;
+}
+
+function expiryDesc(a: ToDoItemTypes, b: ToDoItemTypes) {
+  const first = new Date(a.expireUntil);
+  const second = new Date(b.expireUntil);
+  return first < second ? 1 : -1;
+}
```
Comment (Yuliia143) [inline client/src/redux/selectors/todolistSelectors/index.ts]: I prefer to join it to one function
If you have 10 orders, will you create 10 functions ?

Comment (volodymyryaniv) [reply client/src/redux/selectors/todolistSelectors/index.ts]: Okey,thanks for your feedback.I will fix everything.

Comment (volodymyryaniv) [reply client/src/redux/selectors/todolistSelectors/index.ts]: I hope I understood you correctly, I will correct everything if necessary.

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**